tests/visualization_tests/test_slice.py

```diff
@@ -1,99 +1,364 @@
+    "params",
+    [
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
+        None,
+    ],
+)
+def test_get_slice_plot_info_params(params: Optional[List[str]]) -> None:
+
+    study = prepare_study_with_trials()
+    n_params = len(params) if params is not None else 4
+
+    info = _get_slice_plot_info(study, params=params, target=None, target_name="Objective Value")
+    assert len(info.subplots) == n_params
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+    ],
+)
+def test_get_slice_plot_info_customized_target(params: List[str]) -> None:
+
+    study = prepare_study_with_trials()
+    info = _get_slice_plot_info(
+        study, params=params, target=lambda t: t.params["param_d"], target_name="Objective Value"
+    )
+    n_params = len(params)
+    assert len(info.subplots) == n_params
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        ["param_a", "param_b"],  # First column has 1 observation.
+        ["param_b", "param_a"],  # Second column has 1 observation
+    ],
+)
+def test_generate_contour_plot_for_few_observations(params: List[str]) -> None:
+
+    study = prepare_study_with_trials(less_than_two=True)
+    info = _get_slice_plot_info(study, params, None, "Objective Value")
+
+    assert info == _SlicePlotInfo(
+        target_name="Objective Value",
+        subplots=[
+            _SliceSubplotInfo(
+                param_name="param_a",
+                x=[1.0],
+                y=[0.0],
+                trial_numbers=[0],
+                is_log=False,
+                is_numerical=True,
+            ),
+            _SliceSubplotInfo(
+                param_name="param_b",
+                x=[2.0, 0.0],
+                y=[0.0, 2.0],
+                trial_numbers=[0, 1],
+                is_log=False,
+                is_numerical=True,
+            ),
+        ],
+    )
+
+
+def test_get_slice_plot_info_log_scale_and_str_category_2_params() -> None:
+
+    study = _create_study_with_log_scale_and_str_category_2d()
+    info = _get_slice_plot_info(study, None, None, "Objective Value")
+    assert info == _SlicePlotInfo(
+        target_name="Objective Value",
+        subplots=[
+            _SliceSubplotInfo(
+                param_name="param_a",
+                x=[1e-6, 1e-5],
+                y=[0.0, 1.0],
+                trial_numbers=[0, 1],
+                is_log=True,
+                is_numerical=True,
+            ),
+            _SliceSubplotInfo(
+                param_name="param_b",
+                x=["101", "100"],
+                y=[0.0, 1.0],
+                trial_numbers=[0, 1],
+                is_log=False,
+                is_numerical=False,
+            ),
+        ],
+    )
+
+
+def test_get_slice_plot_info_mixture_category_types() -> None:
+
+    study = _create_study_mixture_category_types()
```

Comment:
How about simply inlining `_create_study_mixture_category_types` function?

Reply:
`test_plot_slice` uses `_create_study_mixture_category_types`.

Reply:
Sorry, I missed that :bow: